tools/localnet/design.md

```diff
@@ -0,0 +1,95 @@
+# LOCALNET
+
+`localnet` is a tool allowing us to set up the entire local ecosystem required to build and test new features.
+
+## Motivation
+
+By providing new features and integrating new blockchains, more and more funds are collected in our ecosystem. It means that
+responsibility of dev team grows every day. That's why we need good tools helping us to build our confidence that new
+feature works well before pull request is even created by the author.
+
+Before IBC era, testing new features locally was pretty easy. The only required component was a single `sifnode`.
+Now, after releasing IBC it is much more complicated, and many interconnected components are needed.
+For ease of use it seems reasonable to create a tool starting and managing all of them.
+
+## Requirements
+
+To be able to play with all th features of `sifnode` (including IBC) we need these components:
+- at least two `sifnodes` running two separate chains
+- relayer between them
+- local eth testnet
+- peggy between eth and sifchain
+- block explorer for every chain
+
+User selects which blockchains are started.
+
+User chooses if block explorers are started.
+
+Every time we add new supported blockchain `localnet` is extended by adding testnet, relayer and block explorer.
+
+### Sifchain
+
+To support sifchains, 4 validators and 1 full node per chain are started.
+
+It would be nice to implement some logs filter in `sifnode` to prevent regular `tendermint` consensus-related logs
+from being printed in `localnet` unless they are errors or panics. By doing this, relevant logs printed by our app are
+more meaningful and take less space on screen.
+
+[`hermes`](https://hermes.informal.systems/) is configured to relay tokens between two sifchains.
+
+[`Big Dipper`](https://github.com/forbole/big-dipper) is used for exploring blocks.
+
+`sifnode` and `hermes` are single-file pieces of software. Both may be run locally easily, so they don't require
+containers. On the other hand `Big Dipper` requires frontend and `mongo` instance, that's why it's better to execute them
+in containers (ready to use images are available there).
+
```

Comment:
Both Sifnode and relayers usually store directories in the user's home folder, so storing this on container can be useful to create isolation.

Reply:
But there is always a CLI flag to modify the path to the config/data dir. I don't see much sense in running software in container for testing purposes because of one directory